logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/conversation/message/ApplicationMessageHandler.kt

```diff
@@ -263,24 +263,60 @@ internal class ApplicationMessageHandlerImpl(
         }
     }
 
+
     private suspend fun handleDeleteMessage(
         content: MessageContent.DeleteMessage,
         conversationId: ConversationId,
         senderUserId: UserId
     ) {
-        if (isSenderVerified(content.messageId, conversationId, senderUserId)) {
-            messageRepository.getMessageById(conversationId, content.messageId).onSuccess { messageToRemove ->
-                (messageToRemove.content as? MessageContent.Asset)?.value?.remoteData?.let { assetToRemove ->
-                    assetRepository.deleteAssetLocally(assetId = assetToRemove.assetId)
-                        .onFailure {
-                            logger.withFeatureId(ApplicationFlow.ASSETS).w("delete messageToRemove asset locally failure: $it")
-                        }
+        messageRepository.getMessageById(conversationId, content.messageId).onSuccess { messageToRemove ->
+            (messageToRemove as? Message.Regular)?.let { message ->
+                if (message.expirationData != null) {
```

Comment:
If we don't add an `else` case, we will break the deletion for regular messages that don't have `expirationData`